### PR TITLE
Imply type of literal returned based on input protobuff for zero elem…

### PR DESCRIPTION
**Pull request by TedThemistokleous**
This saves us the default behavior as the onnx parsing assumes that every zero value is float. This way we're still grabbing relevant type information from the protobuff instead and wont fail our data type checks for if them/else blocks from onnx.

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
Reverted change and uploaded isolated test case for this, looks like we're breaking even if we remove the return the default buffer.

File: test/onnx/gen_onnx.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 
 def onnx_test(op_test):
+
```
Comment (kahmed10) [inline test/onnx/gen_onnx.py:34]: Please make sure to run the same version of yapf as is found on the CI.

File: test/onnx/gen_onnx.py [hunk 2]
```diff
@@ -626,6 +627,46 @@ def constant_scalar_test():
     return ([node], [], [y])
 
 
+@onnx_test
+def constant_scalar_test2():
```
Comment (kahmed10) [inline test/onnx/gen_onnx.py:631]:
```suggestion
def constant_scalar_test_int64():
```

File: test/onnx/gen_onnx.py [hunk 3]
```diff
@@ -626,6 +627,46 @@ def constant_scalar_test():
...
+            dims=x.shape,
+            vals=x.flatten().astype(np.int64),
+        ),
+    )
+
+    return ([node], [], [y])
+
+
+@onnx_test
+def constant_scalar_test3():
```
Comment (kahmed10) [inline test/onnx/gen_onnx.py:651]: This test does not generate a scalar but instead a constant value with 1 element. Can you please rename this?

Comment (TedThemistokleous) [reply test/onnx/gen_onnx.py]: Done

File: test/onnx/gen_onnx.py (hunk 2 above)
Comment (TedThemistokleous) [reply test/onnx/gen_onnx.py]: Done

File: test/onnx/gen_onnx.py (hunk 1 above)
Comment (TedThemistokleous) [reply test/onnx/gen_onnx.py]: Yup. Done

File: test/onnx/gen_onnx.py (hunk 3 above)
Comment (umangyadav) [reply test/onnx/gen_onnx.py]: `constant_scalar_test` is also not generating scalar in that case.

File: test/onnx/gen_onnx.py [hunk 4]
```diff
@@ -626,6 +626,46 @@ def constant_scalar_test():
...
+def constant_one_val_int64_test():
+    x = np.array([1]).astype(np.int64)
+    y = helper.make_tensor_value_info('0', TensorProto.INT64, [0])
+
+    node = onnx.helper.make_node(
+        'Constant',
+        inputs=[],
+        outputs=['0'],
+        value=onnx.helper.make_tensor(
+            name='empty_tensor',
```
Comment (umangyadav) [inline test/onnx/gen_onnx.py:659]: neat picking, it is not empty. Otherwise LGTM.

**Review by umangyadav — APPROVED** [review]

File: test/onnx/gen_onnx.py (hunk 4 above)
Comment (TedThemistokleous) [reply test/onnx/gen_onnx.py:659]: Sure give me a sec let me fix this.

**Review by kahmed10 — APPROVED** [review]

**Comment by TedThemistokleous** [comment]
Fixes #1316

**Pull request merged**